core/src/main/java/org/stellar/anchor/sep6/Sep6Service.java

```diff
@@ -1,38 +1,115 @@
 import org.stellar.anchor.api.exception.*;
 import org.stellar.anchor.api.sep.AssetInfo;
+import org.stellar.anchor.api.sep.SepTransactionStatus;
 import org.stellar.anchor.api.sep.sep6.*;
 import org.stellar.anchor.api.sep.sep6.InfoResponse.*;
 import org.stellar.anchor.api.shared.RefundPayment;
 import org.stellar.anchor.api.shared.Refunds;
 import org.stellar.anchor.asset.AssetService;
 import org.stellar.anchor.auth.Sep10Jwt;
 import org.stellar.anchor.config.Sep6Config;
+import org.stellar.anchor.event.EventService;
+import org.stellar.anchor.util.MemoHelper;
+import org.stellar.anchor.util.SepHelper;
+import org.stellar.anchor.util.TransactionHelper;
+import org.stellar.sdk.KeyPair;
+import org.stellar.sdk.Memo;
 
 public class Sep6Service {
   private final Sep6Config sep6Config;
   private final AssetService assetService;
   private final Sep6TransactionStore txnStore;
+  private final EventService.Session eventSession;
 
   private final InfoResponse infoResponse;
 
   public Sep6Service(
-      Sep6Config sep6Config, AssetService assetService, Sep6TransactionStore txnStore) {
+      Sep6Config sep6Config,
+      AssetService assetService,
+      Sep6TransactionStore txnStore,
+      EventService eventService) {
     this.sep6Config = sep6Config;
     this.assetService = assetService;
     this.txnStore = txnStore;
+    this.eventSession =
+        eventService.createSession(this.getClass().getName(), EventService.EventQueue.TRANSACTION);
     this.infoResponse = buildInfoResponse();
   }
 
   public InfoResponse getInfo() {
     return infoResponse;
   }
 
+  public GetDepositResponse deposit(Sep10Jwt token, GetDepositRequest request)
+      throws AnchorException {
+    // Pre-validation
+    if (token == null) {
+      throw new SepNotAuthorizedException("missing token");
+    }
+    if (request == null) {
+      throw new SepValidationException("missing request");
+    }
+
+    AssetInfo asset = assetService.getAsset(request.getAssetCode());
+    if (asset == null || !asset.getDeposit().getEnabled() || !asset.getSep6Enabled()) {
+      throw new SepValidationException(
+          String.format("invalid operation for asset %s", request.getAssetCode()));
+    }
+
+    try {
+      KeyPair.fromAccountId(request.getAccount());
+    } catch (RuntimeException ex) {
+      throw new SepValidationException(String.format("invalid account %s", request.getAccount()));
+    }
+    Memo memo = MemoHelper.makeMemo(request.getMemo(), request.getMemoType());
+    String id = SepHelper.generateSepTransactionId();
+
+    Sep6TransactionBuilder builder =
+        new Sep6TransactionBuilder(txnStore)
+            .id(id)
+            .transactionId(id)
+            .status(SepTransactionStatus.INCOMPLETE.toString())
+            .kind(Sep6Transaction.Kind.DEPOSIT.toString())
+            .type(request.getType())
+            .assetCode(request.getAssetCode())
+            .assetIssuer(asset.getIssuer())
+            .amountOut(request.getAmount())
```

Comment:
We use amount in jwt in sep-24, should we use amount_requested here?
Because the final amount_out may differ from what user requested

Reply:
`amount_requested` doesn't exist, but I think `amount_expected` is what you meant? If so then yeah, we should use that. I think I made an assumption that business server would amend the `amount_out` when they process the transaction.

Reply:
Yes